recipes/cppcheck/all/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(cmake_wrapper)
+
+include("conanbuildinfo.cmake")
+conan_basic_setup(TARGETS)
+
+# Set include paths
+conan_global_flags()
+
+# Setup libraries to link
+set(Z3_LIBRARIES CONAN_PKG::z3)
+set(PCRE_LIBRARY CONAN_PKG::pcre)
```

Comment:
```suggestion
```

Reply:
Unfortunately this won't work. The z3 package is exposed as z3_LIBRARIES not Z3_LIBRARIES (ref. original Z3Config.cmake.in https://github.com/Z3Prover/z3/blob/master/cmake/Z3Config.cmake.in).

 I could patch more files in cppcheck but I think it is nicer to solve it outside that project. 

As for PCRE I think as cppcheck uses find_library dependencies of PCRE (such as mpir) are not linked in correct? I'm not 100% sure but I remember mpir linker errors.

Reply:
Removing PCRE leads to errors as well.

Reply:
[cppcheck uses find_library to find pcre](https://github.com/danmar/cppcheck/blob/686a1bec99b7721d361eb54356d63d5c8239f565/cmake/findDependencies.cmake#L10), so if you want this to work, you need to add the `cmake_path` generator to the recipe, and add `include(conan_paths.cmake)` to this CMakeLists.txt , as I suggested earlier
Regarding Z3, there is indeed an incompatibility in the varibales generated by conan. this is https://github.com/conan-io/conan/issues/7691 striking again.